### PR TITLE
Add An Option to XUnit.Performance.Analysis to Dump Test Statistics

### PR DESCRIPTION
**Pull request by brianrob**
This option generates a CSV that provides per-test statistics for use in analyzing the effectiveness of the tests themselves.

### CODE REVIEW

**Comment by brianrob** [comment]
@davmason, can you please review this?

File: src/xunit.performance.analysis/Program.cs [hunk 1]
```diff
@@ -318,6 +327,44 @@ private static void WriteTestResultsCSV(Dictionary<string, Dictionary<string, Te
...
+                        RunningStatistics durationStats = result.Stats[DurationMetricName];
+                        writer.WriteLine(
+                            "\"{0}\",\"{1}\",\"{2}\",\"{3}\",\"{4}\",\"{5}\",\"{6}\"",
+                            EscapeCsvString(result.TestName),
+                            durationStats.Count,
+                            durationStats.Minimum,
+                            durationStats.Maximum,
+                            durationStats.Mean,
+                            durationStats.StandardDeviation,
+                            GetMetricsString(result.Stats.Keys)
```
Comment (dsgouda) [inline src/xunit.performance.analysis/Program.cs]: If this is a string, we might want to escape it

**Comment by brianrob** [comment]
@dotnet-bot test this please

File: src/xunit.performance.analysis/Program.cs (hunk 1 above)
Comment (brianrob) [inline src/xunit.performance.analysis/Program.cs]: Which specific field are you thinking should be escaped?

Comment (dsgouda) [inline src/xunit.performance.analysis/Program.cs]: "GetMetricsString(result.Stats.Keys)"  if I understand correctly it should be the metrics which ideally should not have any special characters but just to be on the safer side...

Comment (brianrob) [inline src/xunit.performance.analysis/Program.cs]: Ah, yes... you're right.  Fixed.  Thanks for catching this.

**Comment by davmason** [comment]
Looks good to me

**Comment by brianrob** [comment]
@dotnet-bot test this please

**Comment by brianrob** [comment]
Thanks all for the review.

**Pull request merged**